pkg/server/gitproviders/gitprovider.go

```diff
@@ -19,6 +19,8 @@ func (s *GitProviderService) GetGitProviderForUrl(repoUrl string) (gitprovider.G
 		return nil, "", err
 	}
 
+	var handleableProviders []string
```

Comment:
We can store `gitprovider.GitProvider` here and then just do

```golang
if len(eligibleProviders) > 0 {
 return eligibleProviders[0]
}
```

Actually, looking at this, we don't even need the array.

We can store just one eligible git provider.

```golang
var eligibleProvider *gitprovider.GitProvider

for ... {
 if canHandle {
    eligibleProvider = &gitProvider
}

if eligibleProvider != nil {
  return eligibleProvider
}
```

Reply:
@tarunrajput we should also return the appropriate Id instead of "" here `return eligibleProvider, "", nil`.

Reply:
fixed